### PR TITLE
More complete check before assigning someone a WCA ID.

### PR DESCRIPTION
**Pull request by jfly**
See #1174 and #1173, which only checked for birthdates, but we also care about name, citizenship, and gender. I think that in practice, we'll never have missing names or citizenships, but I don't know if our php admin scripts actually check for that.

Please note, this does not require someone to identify as "male" or "female", that's what our "other" option is for.

### CODE REVIEW

**Comment by pedrosino** [comment]
Funny enough, checks failed :)

File: WcaOnRails/app/models/person.rb [hunk 1]
```diff
@@ -101,6 +101,16 @@ def update_using_sub_id(attributes)
...
   end
 
+  # Note this is very similar to the cannot_register_for_competition_reasons method in user.rb.
+  def cannot_be_assigned_to_user_reasons
+    reasons = []
+    reasons << I18n.t('users.errors.wca_id_no_name_html') if name.blank?
+    reasons << I18n.t('users.errors.wca_id_no_gender_html') if gender.blank?
+    reasons << I18n.t('users.errors.wca_id_no_birthdate_html') if dob.blank?
+    reasons << I18n.t('users.errors.wca_id_no_citizenship_html') if country_iso2.blank?
+    reasons
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/person.rb]: What about using a neat method instead? =)
```ruby
[].tap do |reasons|
  # ...
end
```

File: WcaOnRails/config/locales/en.yml [hunk 2]
```diff
@@ -523,6 +523,9 @@ en:
       senior_has_delegate: "cannot demote senior delegate with subordinate delegates"
       already_have_id: "cannot claim a WCA ID because you already have WCA ID %{wca_id}"
       wca_id_no_birthdate_html: "We do not have a birthdate recorded for this WCA ID. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> with a photo of any legal document (ID card, driver license, etc) where your name and date of birth are visible."
+      wca_id_no_gender_html: "We do not have a gender recorded for this WCA ID. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> to resolve this."
+      wca_id_no_citizenship_html: "We do not have a citizenship recorded for this WCA ID. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> to resolve this."
+      wca_id_no_name_html: "We do not have a name recorded for this WCA ID. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> to resolve this."
```
Comment (jonatanklosko) [inline WcaOnRails/config/locales/en.yml]: The second sentence is identical for each key, could we save our translators some pain? =)

File: WcaOnRails/app/models/person.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/models/person.rb]: Neat! I like that, although the indentation is a bit of a bummer.

File: WcaOnRails/config/locales/en.yml (hunk 2 above)
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: My gut feeling here is that it's not worth making our code more complicated to handle this.

**Pull request merged**